linux-tools/gtk2/gtk2.sh

```diff
@@ -73,13 +73,20 @@ function run_test()
 
 function test-query-modules()
 {
-   tc_register "gtk-query-immodules-2.0-64 testing"
-   gtk-query-immodules-2.0-64 >$stdout 2>$stderr
-   tc_pass_or_fail $? "test failed"
+    tc_get_os_arch
+    if [[ $TC_OS_ARCH == "ppc" || $TC_OS_ARCH == "i686" ]]; then
+        tc_register "gtk-query-immodules-2.0-32 testing"
+        gtk-query-immodules-2.0-32 >$stdout 2>$stderr
+    else
+        tc_register "gtk-query-immodules-2.0-64 testing"
+        gtk-query-immodules-2.0-64 >$stdout 2>$stderr
+    fi
+
```

Comment:
Can this check made generic for any version ?

Reply:
That's a good suggestion but I looked at different versions of gtk2 on different distros. All are providing the same library. So I think, we can consider this as generic one till there is any change.

[root@localhost ~]# rpm -ql gtk2-2.24.28-8.el7.ppc64le | grep gtk-query
/usr/bin/gtk-query-immodules-2.0-64

[root@localhost ~]# rpm -ql gtk2-2.24.31-1.el7.ppc64le | grep gtk-query
/usr/bin/gtk-query-immodules-2.0-64

[root@localhost ~]$ rpm -ql gtk2-2.24.30-1.fc24.i686 | grep gtk-query
/usr/bin/gtk-query-immodules-2.0-32

[root@localhost ~]$ rpm -ql gtk2-2.24.30-1.fc24.x86_64 | grep gtk-query
/usr/bin/gtk-query-immodules-2.0-64